New online account registration

Nagadivyabikkina opened the pull request.

alireza0511 reviewed (commented):
You did a great job, to merge your code, please add it as a feature in hub screen, otherwise your feature blocks others features.

Code review checklist 

- [x] Functionality
                -  Does the code, function as we expect it on iOS?
                -  Does the code, function as we expect it on Android?
- [x]  All test passed locally
- [x]  Architecture
                -  Follow Clean Framework
                -  Follow SOLID principle
- [x]  100% Test Coverage 
- [x]  Readability
                - Easy to read & understand
                - Formatted properly Â 
                - Used proper naming
                - Checked guideline for import section
                - Checked misspelling 
                - Added comment for significant steps and instructions (rest of commented code should be removed.)
- [x]  Reusability
                - Follow the DRY principleÂ and code with no duplication
- [x]  Efficient coding

Nagadivyabikkina commented:
> You did a great job, to merge your code, please add it as a feature in hub screen, otherwise your feature blocks others features.
> 
> Code review checklist
> 
> * [x]  Functionality
>   -  Does the code, function as we expect it on iOS?
>   -  Does the code, function as we expect it on Android?
> * [x]   All test passed locally
> * [x]   Architecture
>   -  Follow Clean Framework
>   -  Follow SOLID principle
> * [x]   100% Test Coverage
> * [x]   Readability
>   - Easy to read & understand
>   - Formatted properly Â
>   - Used proper naming
>   - Checked guideline for import section
>   - Checked misspelling
>   - Added comment for significant steps and instructions (rest of commented code should be removed.)
> * [x]   Reusability
>   - Follow the DRY principleÂ and code with no duplication
> * [x]   Efficient coding

Added new route for Account registration screen and added button for it to push on login screen to avoid conflicts with master and covered unit tests and integration for it too.
